tests/scripts/run-xunit-perf.py

```diff
@@ -0,0 +1,520 @@
+        errorMessage(str): message to print if there is an error
+    """
+    log('')
+    log(" ".join(runArgs))
+
+    try:
+        subprocess.check_output(runArgs, stderr=subprocess.PIPE, env=environment)
+    except subprocess.CalledProcessError as e:
+        log(e.output.decode('utf-8'))
+        raise RuntimeException(errorMessage);
+
+##########################################################################
+# Execution Functions
+##########################################################################
+
+def run_benchmark(benchname, benchdir, env, sandboxDir, benchmarkOutputDir, testFileExt, stabilityPrefix, collectionFlags, lvRunId, isScenarioTest, arch, extension, executable):
+    """ Run perf benchmark
+    Args:
+        benchname (str): name of the benchmark
+        benchdir (srr): path to the benchmark
+        env (str{}): environment for the benchmark
+        sandboxDir (str): path to the sandbox
+        benchmarkOutputDir (str): directory where we should save output of the benchmark
+        testFileExt (str): extension of the benchmark
+        stabilityPrefix (str): stability prefix to run the benchmark under
+        collectionFlags (str): data to collect
+        lvRunId (str): ID of the benchmark
+        isScenarioTest (bool): if the benchmark is a scenario
+        arch (str): arch of the benchmark
+        extension (str): extension for corerun (.exe for windows, empty string for non-windows)
+        executable (str[]): executable to run benchmark under (cmd /c for windows, empty for non-windows)
+
+    Returns: 0 if benchmark passed, 1 if benchmark failed
+    """
+    myEnv = dict(env)
+    benchnameWithExt = benchname + '.' + testFileExt
+    fullPath = os.path.join(benchdir, benchnameWithExt)
+    shutil.copy2(fullPath, sandboxDir)
+
+    files = glob.iglob(os.path.join(benchdir, "*.txt"))
+    for filename in files:
+        if os.path.isfile(filename):
+            shutil.copy2(filename, sandboxDir)
+
+    myEnv['CORE_ROOT'] = sandboxDir
+
+    benchnameLogFileName = os.path.join(benchmarkOutputDir, lvRunId + '-' + benchname + '.log')
+
+    if not os.path.isdir(benchmarkOutputDir):
+        os.makedirs(benchmarkOutputDir)
+
+    log('')
+    log('--------')
+    log('Running %s %s' % (lvRunId, benchname))
+    log('--------')
+    log('')
+
+    lvCommonArgs = [os.path.join(sandboxDir, benchnameWithExt),
+            '--perf:outputdir',
+            benchmarkOutputDir,
+            '--perf:runid',
+            lvRunId]
+
+    if isScenarioTest:
+        lvCommonArgs += ['--target-architecture',
+                arch]
+    else:
+        lvCommonArgs.insert(0, 'PerfHarness.dll')
+
+    splitPrefix = [] if stabilityPrefix is None else stabilityPrefix.split()
+    runArgs = executable + splitPrefix + [os.path.join(sandboxDir, 'corerun' + extension)] + lvCommonArgs + ['--perf:collect', collectionFlags]
+    log(" ".join(runArgs))
+    sys.stdout.flush()
```

Comment:
Just curious what this flush is for?

Reply:
I should remove that, but basically, when it was failing because the quote marks around CORECLR_PERF_RUN were being stripped, the way it was crashing for some reason caused the logging to stop working, so I needed to force the flush. Thanks for the catch!

Reply:
Cool